Build failure: basedpyright

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

See https://hydra.nixos.org/build/308958308

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/308958308

### Relevant log output

```console
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/mq50ispacaymg8xc9iybqr3i5ijqf47v-source
source root is source
Running phase: patchPhase
Executing npmConfigHook
Configuring npm
Validating consistency between /private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/package-lock.json and /nix/store/d2ml7m7b2fc2hjqq2jlgngpms96wx04k-basedpyright-1.31.5-npm-deps/package-lock.json
Setting npm_config_cache to /nix/store/d2ml7m7b2fc2hjqq2jlgngpms96wx04k-basedpyright-1.31.5-npm-deps
Installing dependencies
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated read-package-json@6.0.4: This package is no longer supported. Please use @npmcli/package-json instead.
npm warn deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm warn deprecated npmlog@6.0.2: This package is no longer supported.
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated gauge@4.0.4: This package is no longer supported.
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.

added 1090 packages, and audited 1095 packages in 8s

192 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
patching script interpreter paths in node_modules
node_modules/is-inside-container/node_modules/is-docker/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/is-inside-container/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/jsesc/bin/jsesc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/browserslist/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/jest-junit/node_modules/uuid/dist/bin/uuid: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/flat/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/read-package-json/node_modules/glob/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@detachhead/ts-helpers/node_modules/is-ci/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@vscode/vsce/vsce: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@azure/identity/node_modules/open/xdg-open: interpreter directive changed from "#!/bin/sh" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/sh"
node_modules/@azure/msal-node/node_modules/uuid/dist/bin/uuid: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/prebuild-install/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/import-local/fixtures/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/shx/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/node-addon-api/tools/conversion.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/url-join/bin/changelog: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/rimraf/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/pacote/lib/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/@npmcli/run-script/lib/node-gyp-bin/node-gyp: interpreter directive changed from "#!/usr/bin/env sh" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/sh"
node_modules/npm-check-updates/node_modules/sigstore/bin/sigstore.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/node-gyp/update-gyp.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/bin/node-gyp.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/node-gyp/node_modules/rimraf/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/node-gyp/node_modules/which/bin/node-which: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/gyp_main.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/tools/pretty_sln.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/tools/pretty_vcproj.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/tools/graphviz.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/tools/emacs/run-unit-tests.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/sh"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/tools/pretty_gyp.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/common_test.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/win_tool.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/input_test.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/__init__.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/flock_tool.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/generator/msvs_test.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/MSVSSettings_test.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/mac_tool.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/pylib/gyp/easy_xml_test.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/test_gyp.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
node_modules/npm-check-updates/node_modules/node-gyp/gyp/gyp: interpreter directive changed from "#!/bin/sh" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/sh"
node_modules/npm-check-updates/node_modules/glob/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/which/bin/which.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/node_modules/nopt/bin/nopt.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/build/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/npm-check-updates/build/src/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/rimraf/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/resolve/bin/resolve: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/parse-semver/node_modules/semver/bin/semver: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/jest-cli/bin/jest.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/acorn/bin/acorn: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/prettier/bin-prettier.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/babel-plugin-istanbul/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ts-jest/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ts-jest/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/miller-rabin/bin/miller-rabin: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/jake/bin/bash_completion.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/bash"
node_modules/jake/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/vscode-languageclient/lib/node/terminateProcess.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/bash"
node_modules/typescript/bin/tsserver: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/typescript/bin/tsc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/is-ci/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/shelljs/bin/shjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/js-yaml/bin/js-yaml.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ejs/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ts-command-line-args/dist/write-markdown.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/esprima/bin/esparse.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/esprima/bin/esvalidate.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/webpack-cli/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-lint-semver-ranges/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-list-mismatches/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-format/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-lint/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-fix-mismatches/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-prompt/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-set-semver-ranges/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-update/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/dist/bin-list/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/syncpack/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/pyright-to-gitlab-ci/dist/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/webpack/bin/webpack.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@npmcli/installed-package-contents/bin/index.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/envinfo/dist/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/markdown-it/bin/markdown-it.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@babel/core/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@babel/parser/bin/babel-parser.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@babel/helper-compilation-targets/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ovsx/node_modules/is-ci/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/ovsx/lib/ovsx: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/json5/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/parse-github-url/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/update-browserslist-db/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/exit/test/fixtures/create-files.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/bash"
node_modules/cross-env/src/bin/cross-env-shell.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/cross-env/src/bin/cross-env.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/mime/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/mime/src/build.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/eslint-config-prettier/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/eslint/bin/eslint.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/sha.js/bin.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/vscode-languageserver/bin/installServerIntoExtension: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/esbuild/bin/esbuild: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/terser/bin/uglifyjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/terser/bin/terser: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/@istanbuljs/load-nyc-config/node_modules/js-yaml/bin/js-yaml.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/rc/cli.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/create-jest/bin/create-jest.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/color-support/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/which/bin/node-which: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/mkdirp/bin/cmd.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
node_modules/jest/bin/jest.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node"
npm error code 1
npm error path /private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar
npm error command failed
npm error command sh -c prebuild-install || npm run build
npm error > keytar@7.9.0 build
npm error > node-gyp rebuild
npm error
npm error make: Entering directory '/private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar/build'
npm error   CXX(target) Release/obj.target/keytar/src/async.o
npm error make: Leaving directory '/private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar/build'
npm error prebuild-install warn install EACCES: permission denied, access '/nix/store/d2ml7m7b2fc2hjqq2jlgngpms96wx04k-basedpyright-1.31.5-npm-deps'
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@11.2.0
npm error gyp info using node@22.19.0 | darwin | arm64
npm error gyp info find Python using Python version 3.13.7 found at "/nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3"
npm error gyp WARN read config.gypi ENOENT: no such file or directory, open '/nix/store/ck8mkvjga2m0b1lqj675g9xf3z19bian-nodejs-22.19.0-source/include/node/config.gypi'
npm error gyp info spawn /nix/store/xjpv7j44jn7mifw8r69p7shrsh1aqmnf-python3-3.13.7/bin/python3
npm error gyp info spawn args [
npm error gyp info spawn args '/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm error gyp info spawn args 'binding.gyp',
npm error gyp info spawn args '-f',
npm error gyp info spawn args 'make',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar/build/config.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/nix/store/ck8mkvjga2m0b1lqj675g9xf3z19bian-nodejs-22.19.0-source/common.gypi',
npm error gyp info spawn args '-Dlibrary=shared_library',
npm error gyp info spawn args '-Dvisibility=default',
npm error gyp info spawn args '-Dnode_root_dir=/nix/store/ck8mkvjga2m0b1lqj675g9xf3z19bian-nodejs-22.19.0-source',
npm error gyp info spawn args '-Dnode_gyp_dir=/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp',
npm error gyp info spawn args '-Dnode_lib_file=/nix/store/ck8mkvjga2m0b1lqj675g9xf3z19bian-nodejs-22.19.0-source/$(Configuration)/node.lib',
npm error gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar',
npm error gyp info spawn args '-Dnode_engine=v8',
npm error gyp info spawn args '--depth=.',
npm error gyp info spawn args '--no-parallel',
npm error gyp info spawn args '--generator-output',
npm error gyp info spawn args 'build',
npm error gyp info spawn args '-Goutput_dir=.'
npm error gyp info spawn args ]
npm error gyp info spawn make
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error In file included from ../src/async.cc:4:
npm error ../../node-addon-api/napi.h:1147:39: error: in-class initializer for static data member is not a constant expression
npm error  1147 |     static const napi_typedarray_type unknown_array_type = static_cast<napi_typedarray_type>(-1);
npm error       |                                       ^                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error ../../node-addon-api/napi.h:1147:60: note: integer value -1 is outside the valid range of values [0, 15] for the enumeration type 'napi_typedarray_type'
npm error  1147 |     static const napi_typedarray_type unknown_array_type = static_cast<napi_typedarray_type>(-1);
npm error       |                                                            ^
npm error 1 error generated.
npm error make: *** [keytar.target.mk:126: Release/obj.target/keytar/src/async.o] Error 1
npm error gyp ERR! build error 
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:219:23)
npm error gyp ERR! System Darwin 24.6.0
npm error gyp ERR! command "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/bin/node" "/nix/store/7hnbd3z2flnqh17fh3sbk0srpn6zvzck-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm error gyp ERR! cwd /private/tmp/nix-build-basedpyright-1.31.5.drv-0/source/node_modules/keytar
npm error gyp ERR! node -v v22.19.0
npm error gyp ERR! node-gyp -v v11.2.0
npm error gyp ERR! not ok
npm error Log files were not written due to an error writing to the directory: /nix/store/d2ml7m7b2fc2hjqq2jlgngpms96wx04k-basedpyright-1.31.5-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
```

### Additional context

@r-ryantm , I run a review in your PR #447322

I think should let hydra to build again

### System metadata

- hydra

### Notify maintainers

@kiike @MisileLab 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ba/basedpyright/package.nix
+++ b/pkgs/by-name/ba/basedpyright/package.nix
@@ -2,6 +2,8 @@
   lib,
   fetchFromGitHub,
   runCommand,
+  stdenv,
+  clang_20,
   buildNpmPackage,
   docify,
   testers,
@@ -37,7 +39,8 @@ buildNpmPackage rec {
   nativeBuildInputs = [
     docify
     pkg-config
-  ];
+  ]
+  ++ lib.optional stdenv.isDarwin [ clang_20 ]; # clang_21 breaks keytar
 
   buildInputs = [ libsecret ];
 

